gateways/stripe/stripe_vars.php

```diff
@@ -2,17 +2,58 @@
 
 function espresso_display_stripe($payment_data) {
 	extract($payment_data);
-	global $org_options;
+	global $org_options, $wpdb;
 	$stripe_settings = get_option('event_espresso_stripe_settings');
 	if ($stripe_settings['force_ssl_return']) {
-			$home = str_replace('http://', 'https://', home_url());
-		} else {
-			$home = home_url();
-		}
+		$home = str_replace('http://', 'https://', home_url());
+	} else {
+		$home = home_url();
+	}
+	
+	$SQL = "SELECT start_date FROM " . EVENTS_ATTENDEE_TABLE . " WHERE registration_id=%s ORDER BY id LIMIT 1";
+	$payment_data['start_date'] = $wpdb->get_var( $wpdb->prepare( $SQL, $payment_data['registration_id']));
 
-	wp_register_script( 'stripe', EVENT_ESPRESSO_PLUGINFULLURL . 'gateways/stripe/stripe.js', array( 'jquery.validate.js' ), '1.0', TRUE );
-	wp_enqueue_script( 'stripe' );	
+	$stripe_description = sprintf(
+		/* translators: 1: event name, 2: event date, 3: attendee first name, 4: attendee last name */
+		__('%1$s [%2$s] >> %3$s %4$s', 'event_espresso'),
+		$payment_data["event_name"],
+		date('m-d-Y', strtotime($payment_data['start_date'])),
+		$payment_data["fname"],
+		$payment_data["lname"]
+	);
+	
+	//Check for an alternate Stripe settings
+	if ( !empty($payment_data['event_meta']['stripe_publishable_key'] ) ) {
+		//Alternate Stripe settings
+		$stripe_settings['stripe_publishable_key'] = $payment_data['event_meta']['stripe_publishable_key'];
+	}
 
+	wp_enqueue_script('stripe_payment_js', 'https://js.stripe.com/v3/', array(), false, true);
+	wp_enqueue_script('espresso_stripe', EVENT_ESPRESSO_PLUGINFULLURL . 'gateways/stripe/espresso_stripe.js', array('stripe_payment_js'), false, true);
+	wp_enqueue_style('espresso_stripe_css', EVENT_ESPRESSO_PLUGINFULLURL . 'gateways/stripe/espresso_stripe.css');
+
+	//Include the Stripe API
+	require_once (dirname(__FILE__).'/stripe-php-6.43.1/init.php');
+	// Setup a Stripe object using the secret key.
+	\Stripe\Stripe::setApiKey( $stripe_settings['stripe_secret_key']);
+
+	$payment_intent_id = isset($_REQUEST['espresso_stripe_payment_intent_id']) ? $_REQUEST['espresso_stripe_payment_intent_id'] : '';
```

Comment:
should sanitize that ID coming from the `$_REQUEST` array if it's not being sanitized elsewhere

Reply:
So, I had that from a different route I'd tried and in the end just left it in as I didn't see how it could hurt, as its causing confusion already I'll remove it.

The value from `$_REQUEST['espresso_stripe_payment_intent_id']` is only sent to Stripe themselves to pull in the PI object and then the object is used elsewhere but I can sanitize it.